ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -35,19 +41,29 @@
           .put(SqlBaseType.DOUBLE, Number::doubleValue)
           .build();
 
-  public <T> Optional<T> coerce(final Object value, final SqlType targetType) {
-    if (targetType.baseType() == SqlBaseType.ARRAY
-        || targetType.baseType() == SqlBaseType.MAP
-        || targetType.baseType() == SqlBaseType.STRUCT
-    ) {
+  @Override
+  public Optional<Object> coerce(final Object value, final SqlType targetType) {
```

Comment:
Isn't this strictly incorrect generics?  

The method is not returning an optional containing an instance of `Object`, but a subclass of `Object`.  So strictly speaking it should be `Optional<? extends Object>`, which is more succinctly put as `Optional<?>`, i.e. it's returning an `Optional` of an unknown type.

Same goes for other return values...